File: frontend/src/app/types/dto/person/related/contact/mail/person-email-address.dto.ts
import { EmailAddressDto } from '../../../../email-address.dto';
import { PersonEmailAddressDtoId } from '../../reference/contact/person-email-address-dto-id';


export interface PersonEmailAddressDto {

    id?: PersonEmailAddressDtoId;

    emailAddress: EmailAddressDto;

    mainEmail: boolean;

}

File: backend/src/main/java/ibee/webapp/todo_app/features/person/related/contact/dto/PersonEmailAddressDto.java
package ibee.webapp.todo_app.features.person.related.contact.dto;

import ibee.webapp.todo_app.dto.EmailAddressDto;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonEmailAddressDtoId;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public record PersonEmailAddressDto(

    @NotNull
    @Valid 
    PersonEmailAddressDtoId id,

    @NotNull
    @Valid 
    EmailAddressDto emailAddress,

    @NotNull
    Boolean mainEmail
) {

}
